docs/pywbemlistener/commands.rst

```diff
@@ -0,0 +1,240 @@
+..
+.. Licensed under the Apache License, Version 2.0 (the "License");
+.. you may not use this file except in compliance with the License.
+.. You may obtain a copy of the License at
+..
+..    http://www.apache.org/licenses/LICENSE-2.0
+..
+.. Unless required by applicable law or agreed to in writing, software
+.. distributed under the License is distributed on an "AS IS" BASIS,
+.. WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. See the License for the specific language governing permissions and
+.. limitations under the License.
+..
+
+
+.. _`Pywbemlistener commands`:
+
+Pywbemlistener commands
+=======================
+
+This section describes each of the pywbemlistener commands including examples.
+
+The pywbemlistener commands are:
+
+* :ref:`pywbemlistener list command` - List the currently running named WBEM indication listeners.
+* :ref:`pywbemlistener show command` - Show a named WBEM indication listener.
+* :ref:`pywbemlistener start command` - Start a named WBEM indication listener in the background.
+* :ref:`pywbemlistener stop command` -  Stop a named WBEM indication listener.
+* :ref:`pywbemlistener run command` - Run as a named WBEM indication listener.
```

Comment:
How did we end up with an order where start and run are not adjacent. in this list.. I think the order should be run, start, stop

Reply:
run is the one we don't want users to use directly, that's why I put it at the end.

Reply:
Except that on the general help it shows up as the second on the list with pywbemlistener -h

```
  list   List the currently running named WBEM indication listeners.
  run    Run as a named WBEM indication listener.
  show   Show a named WBEM indication listener.
  start  Start a named WBEM indication listener in the background.
  stop   Stop a named WBEM indication listener.

As the help, etc. works, That is the one the user would pick first.   Is it that the codes is such that we absoluteily do not want the user to execute the run or just that it creates a dependent process that can only be killed by ctrl-c or something similar?  If we absolutely do not want them to use run, I think we should hide the command. 

 Click has a hidden option on the commands class that I think hides the help but allows the command to run.  Would that be a solution???

Reply:
I still hang on to the possibility that users have their own mechanisms for using the run command. For example, it could be turned into an OS service or a job. So I would not want to hide the run command.

The order of commands in the general help is alphabetical. Can that be influenced?

Reply:
Let me look at the ordering.  At one point we had set the order as the order in the _cmd)....py file but I think we created a subclass that sorted the order in our click customizations.

Reply:
DONE
* Used move_to_end on click.group() to move the run command to the end of the help list.
* Added the following text to the help text for the run command: 
  "Note: The `pywbemlistener start` command should be used to start listeners,
    and it starts a `pywbemlistener run` command as a background process.
    Use the `pywbemlistener run` command only when you need to have control
    over how exactly the process runs in the background."